src/shortcuts/newThoughtOrOutdent.tsx

```diff
@@ -35,12 +35,12 @@ const exec: Shortcut['exec'] = (dispatch, getState, e, { type }: { type: string
     const target = e.target as HTMLElement
 
     // Note: e.target should be a HTMLElement and a content editable node
-    const offset = cursor && isTargetHTMLElement && target.hasAttribute('contenteditable')
-      ? getOffsetWithinContent(target)
-      : 0
+    const isTargetAnEditable = isTargetHTMLElement && target.hasAttribute('contenteditable')
+
+    const splitResult = cursor && isTargetAnEditable ? splitAtSelection(target) : null
```

Comment:
Just for my understanding, does `splitResult.previous + splitResult.next` always equal `target.innerHTML`?

Reply:
Not always. So if we have following value at the target.innerHTML `<b> bold and <i> italic <i> </b>`.

So splitting with cursor selection right after the word `and`, we will get following result.

`splitResult.previous` will be `<b> bold and </b>`
`splitResult.next` will be `<b><i> italic </i></b>`

We are basically using browser to do the splitting 😄